packages/react/src/internal/useMatchMedia.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useState, useEffect } from 'react';
+import { canUseDOM } from '../../internal/environment';
+
+export function useMatchMedia(mediaQueryString) {
```

Comment:
If you can add tests for this, that'd be great. There's a good example from: https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom that talks about it

We basically would want to test that:

- The initial state of the hook matches if the `mediaQueryList` matches
- When `change` is fired, the hook returns a value that matches what the change event matches
- When the `mediaQueryString` is changed, the state is kept in sync so that the returned value matches the new `mediaQueryList` matches value